In [1]:
import pandas as pd
s=pd.read_csv('C:/Users/Pradeep/Downloads/TEST/drug_review_dataset_with_sentiment (1).csv')
s1=pd.DataFrame(s)
s1.head()

Unnamed: 0                  drugName                     condition  \
0      206461                 Valsartan  Left Ventricular Dysfunction   
1       95260                Guanfacine                          ADHD   
2       92703                    Lybrel                 Birth Control   
3      138000                Ortho Evra                 Birth Control   
4       35696  Buprenorphine / naloxone             Opiate Dependence   

                                              review  rating  \
0  "It has no side effect, I take it in combinati...     9.0   
1  "My son is halfway through his fourth week of ...     8.0   
2  "I used to take another oral contraceptive, wh...     5.0   
3  "This is my first time using any form of birth...     8.0   
4  "Suboxone has completely turned my life around...     9.0   

                date  usefulCount   drug_class  sentiment sentiment_label  
0       May 20, 2012           27  arb blocker   0.000000         neutral  
1     April 27, 2010          192          NaN   0.168333        positive  
2  December 14, 2009           17          NaN   0.067210        positive  
3   November 3, 2015           10          NaN   0.179545        positive  
4  November 27, 2016           37          NaN   0.194444        positive

In [2]:
a=s1['drugName'][s1['rating'].max()]

#### 1

In [3]:
a

'Topiramate'

#### 2

In [4]:
s1['drug_class'].value_counts()

antifungal (except metronidazole)    4201
opiod analgesics                     3945
oral hypoglycemics                   3555
antibiotic                           3401
anti-anxiety                         2645
h2 blockers (anti-ulcers)            1228
beta blockers                         966
corticosteroid (prednisone)           886
antipyschotics (phenothiazine)        664
arb blocker                           560
ace inhibitor                         432
calcium channel blocker               233
alpha blocker                         153
anesthetic                            129
antibiotic(penicillins)               119
thrombolytics                         116
beta blocker                           97
neuromuscular blocking agents          45
antibiotic (cephalosporins)            29
pituitary hormone                      28
barbiturate                            19
anticoagulants                          9
Name: drug_class, dtype: int64

#### 3

In [5]:
b=s1['drugName'][s1['rating']>8]
b

0                        Valsartan
4         Buprenorphine / naloxone
7                     Aripiprazole
10                      Topiramate
11                  L-methylfolate
                    ...           
161284                 Venlafaxine
161287                      Ativan
161292                     Campral
161295          Thyroid desiccated
161296                Lubiprostone
Name: drugName, Length: 78520, dtype: object

#### 4

In [6]:
s1['drugName'].value_counts()

Levonorgestrel                       3657
Etonogestrel                         3336
Ethinyl estradiol / norethindrone    2850
Nexplanon                            2156
Ethinyl estradiol / norgestimate     2117
                                     ... 
Dynacin                                 1
Clioquinol / hydrocortisone             1
Tums Regular Strength                   1
Cyclopentolate                          1
Striant                                 1
Name: drugName, Length: 3436, dtype: int64

We have 3436 types of drugs

#### 5

In [7]:
s1['condition'].value_counts()

Birth Control                28788
Depression                    9069
Pain                          6145
Anxiety                       5904
Acne                          5588
                             ...  
llicular Lymphoma                1
Parkinsonism                     1
ibrocystic Breast Disease        1
ungal Pneumonia                  1
Infectious Diarrhea              1
Name: condition, Length: 884, dtype: int64

From the above data,we have approximtely 4 drugs for each condition

#### 6

In [8]:
s1['usefulCount'].value_counts()

2      6391
3      6379
0      6318
4      6258
1      6148
       ... 
340       1
730       1
337       1
400       1
585       1
Name: usefulCount, Length: 389, dtype: int64

In [16]:
s1['usefulCount'].median()

16.0

16 patients are tested on particular drug

#### 8

In [22]:
s1['sentiment_label'].value_counts()

positive    101041
negative     53303
neutral       6953
Name: sentiment_label, dtype: int64

#### 9

In [23]:
df=s1[['rating','usefulCount']]
df.corr()

rating  usefulCount
rating       1.000000     0.234185
usefulCount  0.234185     1.000000

#### 10

Its ambiguous to depict the rating using the review.

#### 11

In [25]:
s1['rating'].unique()

array([ 9.,  8.,  5.,  2.,  1., 10.,  4.,  3.,  7.,  6.])

Rating is distributed in the scale of 1 to 10.

#### 12

In [32]:
s1['year']=s1['date'].str.split(',').str[1]
s1['year']=s1['year'].astype(int)

In [35]:
s1['year'].value_counts().mean()

16129.7

Amount of reviews per year is in average of 16130

In [38]:
s1['month']=s1['date'].str.split(' ').str[0]
s1['month'].value_counts().mean()

13441.416666666666

Amount of reviews per month is in average of 13441

#### 13